cl/lib/recap_utils.py

```diff
@@ -175,40 +177,69 @@ def needs_ocr(content):
     Case: 08-9007   Document: 00115928542   Page: 2   Date Filed: 07/30/2009   Entry ID: 5364336
     Line 1
 
+    - Character-per-page threshold: If `page_count` is provided,
+    normalize the content (remove extra spaces/newlines) and divide the
+    character count by the number of pages. If the result is less than
+    `CHARS_THRESHOLD_OCR_PER_PAGE`, OCR is needed.
 
     As a fallback it removes these common headers so that if no text remains,
     we can be sure that the PDF needs OCR.
 
-    :param content: The content of a PDF.
-    :return: boolean indicating if OCR is needed.
+    :param content: The text content to check if OCR is needed.
+    :param page_count: The number of pages of the document.
+    :return: True if OCR is needed, False otherwise.
     """
     lines = (ln.strip() for ln in content.splitlines())
     in_page = False
-    other_content_count = 0
+    other_content_chars_count = 0
     saw_any_page = False
+    content_in_page = ""
     for line in lines:
         if is_page_line(line):
             if (
                 in_page
-                and other_content_count < settings.LINE_THRESHOLD_OCR_PER_PAGE
+                and other_content_chars_count
+                < settings.CHARS_THRESHOLD_OCR_PER_PAGE
+                and "Exhibit" not in content_in_page
             ):
+                # If we reached the end of a page and the amount of non-header
+                # content is below the configured threshold, then this page
+                # likely needs OCR.
+                # Exception: if the page contains the word "Exhibit", we assume
+                # it may be a valid exhibit cover page with little text, so we
+                # do not flag it for OCR.
                 return True
             in_page = True
             saw_any_page = True
-            other_content_count = 0
+            other_content_chars_count = 0
```

Comment:
We should also reset the `content_in_page` variable here. I noticed that when a document includes an '**Exhibit**' cover page, the word '**Exhibit**' remains in this variable along with the content from other pages since it’s never cleared.

Reply:
Thanks. You're completely right. I've updated the code and added an additional test to cover this issue.